client/opentracing/src/main/java/io/opentracing/APMSpanBuilder.java

```diff
@@ -34,20 +34,24 @@
 
     private TraceRecorder recorder;
     private ContextSampler sampler;
+    private String serviceName;
+    private String buildStamp;
 
     /**
      * @param operationName The operation name
      * @param recorder The trace recorder
      */
-    APMSpanBuilder(String operationName, TraceRecorder recorder, ContextSampler sampler) {
+    APMSpanBuilder(String operationName, TraceRecorder recorder, ContextSampler sampler, String serviceName, String buildStamp) {
```

Comment:
What about a new API that provides a typed interface for setting the supported properties, so we just add another setter/getter when we want to add another property?